metrics/graphite/graphite.go

```diff
@@ -0,0 +1,295 @@
+// Package graphite implements a graphite backend for package metrics.
+//
+// The current implementation ignores fields.
+package graphite
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"math"
+	"net"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/go-kit/kit/metrics"
+)
+
+// Emitter will keep track of all metrics and, once started,
+// will emit the metrics via the Flush method to the given address.
+type Emitter interface {
+	NewCounter(string) metrics.Counter
+	NewHistogram(string, int64, int64, int, ...int) metrics.Histogram
+	NewTimeHistogram(string, time.Duration, int64, int64, int, ...int) metrics.TimeHistogram
+	NewGauge(string) metrics.Gauge
+
+	Start(time.Duration)
+	Flush() error
+}
+
+type emitter struct {
+	addr   *net.TCPAddr
+	prefix string
+
+	metricMu   *sync.Mutex
```

Comment:
- Canonical Go kit style is to call this `mtx`
- The non-pointer type `sync.Mutex` has a usable default value and should probably be preferred — this is fine as you're (correctly) defining methods on `*emitter`